client/src/app/core/core-services/data-store.service.ts

```diff
@@ -278,6 +278,8 @@ export class DataStoreService {
      */
     private readonly clearEvent: EventEmitter<void> = new EventEmitter<void>();
```

Comment:
please merge this event with the clear event: Just one clear event, parametrized per collection.